h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -23,10 +24,143 @@ class SearchBarController extends Controller {
     this._lozengeContainer = this.refs.searchBarLozenges;
 
 
+    /**
+     * the suggestionsMap pulls in the available lists - either
+     *  static or dynamic living in the dom - into one mapping
+     *  lists of all suggestion values.
+     */
+    this._suggestionsMap = (()=>{
+
+      let explanationList = [
+        {
+          matchOn: 'user',
+          title: 'user:',
+          explanation: 'search by username',
+        },
+        {
+          matchOn: 'tag',
+          title: 'tag:',
+          explanation: 'search for annotations with a tag',
+        },
+        {
+          matchOn: 'url',
+          title: 'url:',
+          explanation: 'see all annotations on a page',
+        },
+        {
+          matchOn: 'group',
+          title: 'group:',
+          explanation: 'show annotations created in a group you are a member of',
+        },
+      ].map((item)=>{ return Object.assign(item, { type: FACET_TYPE}); });
+
+      // tagSuggestions are made available by the scoped template data.
+      // see search.html.jinja2 for definition
+      const tagSuggestionJSON = document.querySelector('.js-tag-suggestions');
+      let tagSuggestions = [];
+
+      if(tagSuggestionJSON){
+        try{
+          tagSuggestions = JSON.parse(tagSuggestionJSON.innerHTML.trim());
+        }catch(e){
+          console.error('Could not parse .js-tag-suggestions JSON content', e);
+        }
+      }
+
+      let tagsList = ((tagSuggestions) || []).map((item)=>{
+        return Object.assign(item, {
+          type: TAG_TYPE,
+          title: item.tag, // make safe
+          matchOn: stringUtil.fold(stringUtil.normalize(item.tag)),
+          usageCount: item.count || 0,
+        });
+      });
+
+      // groupSuggestions are made available by the scoped template data.
+      // see search.html.jinja2 for definition
+      const groupSuggestionJSON = document.querySelector('.js-group-suggestions');
+      let groupSuggestions = [];
+
+      if(groupSuggestionJSON){
+        try{
+          groupSuggestions = JSON.parse(groupSuggestionJSON.innerHTML.trim());
+        }catch(e){
+          console.error('Could not parse .js-group-suggestions JSON content', e);
+        }
+      }
+
+      let groupsList = ((groupSuggestions) || []).map((item)=>{
+        return Object.assign(item, {
+          type: GROUP_TYPE,
+          title: item.name, // make safe
+          matchOn: stringUtil.fold(stringUtil.normalize(item.name)),
+          pubid: item.pubid,
+          name: item.name,
+        });
+      });
+
+      return explanationList.concat(tagsList, groupsList);
+    })();
+
     var getTrimmedInputValue = () => {
       return this._input.value.trim();
     };
 
+
+    /**
+     * given a lozenge value for a group: find the pubid, if we can
```

Comment:
I found it difficult to parse this comment and understand what the inputs/outputs of this function are.

My understanding reading through the code is that it accepts user input of the form `group:{pubid|name}` and returns `{ input: "group:{pubid}", display: "group:{name}" }` if the group with the corresponding pubid or name is found or `{ input: <original input>, display: <original input> }` otherwise.